lib/internal/streams/writable.js

```diff
@@ -749,9 +757,11 @@ function finish(stream, state) {
   state.pendingcb--;
   state.finished = true;
 
-  const onfinishCallbacks = state[kOnFinished].splice(0);
-  for (let i = 0; i < onfinishCallbacks.length; i++) {
-    onfinishCallbacks[i](null);
+  if (state[kOnFinished]) {
+    const onfinishCallbacks = state[kOnFinished].splice(0);
```

Comment:
Why do we need to copy this? We can iterate it once and later assign it to `[]`?

Reply:
He callbacks can modify it